src/model.ts

```diff
@@ -7,67 +7,123 @@ import { IContentFieldFilter, IContentSearchRequest } from '@esri/hub-search';
+    limit?: number
+  }
+}
 
 export class HubApiModel {
 
-  async getStream (request: Request) {
-    const searchRequest: IContentSearchRequest = request.res.locals.searchRequest;
-    this.preprocessSearchRequest(searchRequest);
+  async getStream(request: Request) {
+   
+    const { res: { locals: { searchRequestBody, siteModel, arcgisPortal, siteIdentifier } }, query: { limit } }: HubApiRequest = request;
+    const hubSiteModel = siteModel || await this.fetchHubSiteModel(siteIdentifier, this.getRequestOptions(arcgisPortal));
+    this.preprocessSearchRequest(searchRequestBody);
 
-    if (searchRequest.options.fields) {
+    if (searchRequestBody.options.fields) {
+      searchRequestBody.options.fields = this.getValidHubApiFields(searchRequestBody.options.fields);
       const validFields: string[] = await hubApiRequest('/fields');
-      this.validateFields(searchRequest, validFields);
+      this.validateFields(searchRequestBody, validFields);
     }
 
     // Only fetch site if site is provided and either group or orgid is undefined
     if (
-      searchRequest.options.site &&
+      searchRequestBody.options.site &&
       (
-        !this.scopedFieldValueIsValid(searchRequest.filter.group) || 
-        !this.scopedFieldValueIsValid(searchRequest.filter.orgid)
+        !this.scopedFieldValueIsValid(searchRequestBody.filter.group) ||
+        !this.scopedFieldValueIsValid(searchRequestBody.filter.orgid)
       )
     ) {
-      const siteCatalog = await this.getSiteCatalog(searchRequest, searchRequest.options.site);
-      if (!this.scopedFieldValueIsValid(searchRequest.filter.group)) {
-        searchRequest.filter.group = siteCatalog.groups;
+      const siteCatalog = await this.getSiteCatalog(searchRequestBody, searchRequestBody.options.site);
+      if (!this.scopedFieldValueIsValid(searchRequestBody.filter.group)) {
+        searchRequestBody.filter.group = siteCatalog.groups;
       }
-      if (!this.scopedFieldValueIsValid(searchRequest.filter.orgid)) {
-        searchRequest.filter.orgid = siteCatalog.orgId;
+      if (!this.scopedFieldValueIsValid(searchRequestBody.filter.orgid)) {
+        searchRequestBody.filter.orgid = siteCatalog.orgId;
       }
     }
 
     // Validate the scope to ensure that a group, org, and/or id are present to avoid
     // scraping entire database
-    this.validateRequestScope(searchRequest);
-    
-    const pagingStreams: PagingStream[] = await getBatchedStreams(searchRequest, Number(request.query?.limit));
-    const pass: PassThrough =  new PassThrough({ objectMode: true });
-    return searchRequest.options.sortField 
+    this.validateRequestScope(searchRequestBody);
+
+    const searchRequestWithRequiredFields : IContentSearchRequest = _.cloneDeep(searchRequestBody);
+    searchRequestWithRequiredFields.options.fields =
+      searchRequestBody.options.fields 
+        ? `${searchRequestBody.options.fields},${REQUIRED_FIELDS.join(',')}` : 
+          REQUIRED_FIELDS.join(',');
```

Comment:
move building of searchRequestWithRequiredFields to a helper.

Reply:
moved this to helper function called `searchRequestWithRequiredFields`